lib/OpenQA/Worker/Job.pm

```diff
@@ -660,35 +662,37 @@ sub _upload_results {
     my $is_final_upload = $self->is_stopped_or_stopping;
     $self->{_is_uploading_results} = 1;
 
-    # query status from isotovideo (unless it is the final status upload because isotovideo
-    # has already been stopped in this case)
-    return Mojo::IOLoop->next_tick(sub { $self->_upload_results_step_0_prepare($is_final_upload, {}, $callback) })
-      if $is_final_upload;
-
-    $self->isotovideo_client->status(
-        sub {
-            my ($isotovideo_client, $status_from_os_autoinst) = @_;
-            $self->_upload_results_step_0_prepare($is_final_upload, $status_from_os_autoinst, $callback);
-        });
+    $self->_upload_results_step_0_prepare($is_final_upload, $callback);
 }
 
 sub _upload_results_step_0_prepare {
-    my ($self, $is_final_upload, $status_from_os_autoinst, $callback) = @_;
+    my ($self, $is_final_upload, $callback) = @_;
 
     my $worker_id       = $self->client->worker_id;
     my $job_url         = $self->isotovideo_client->url;
     my $global_settings = $self->worker->settings->global_settings;
+    my $pooldir         = $self->worker->pool_directory;
+    my $status_file     = "$pooldir/" . AUTOINST_STATUSFILE;
     my %status          = (
         worker_id             => $worker_id,
         cmd_srv_url           => $job_url,
         worker_hostname       => $global_settings->{WORKER_HOSTNAME},
-        test_execution_paused => $status_from_os_autoinst->{test_execution_paused} // 0,
+        test_execution_paused => 0,
     );
 
+    my $test_status = {};
+    if (-r $status_file) {
+        $test_status = Cpanel::JSON::XS->new->decode(path($status_file)->slurp);
```

Comment:
Just noticed that this class already imports `decode_json`, that shortens the code even more. 😉

Reply:
(just in case you didn't know, `Mojo::JSON` uses `Cpanel::JSON::XS` automatically when available)

Reply:
oh, and I realized that I never added a `use Cpanel::JSON::XS`